### DISCUSSION TITLE
Hotspots sometimes not hidden behind solid parts of model

### DISCUSSION THREAD

**Discussion by alansegar** [google/model-viewer#1932] [Q&A]
### Description

**Expected behaviour:** Hotspots change to hidden when a solid part of the model blocks the line of sight from the camera to the hotspot
**Actual behaviour:** Sometimes hotspots remain visible despite a solid part of the model blocking the view of the part of the model where the hotspot is located.
The behaviour seems to vary between models and the direction of view. Simpler models e.g. one large hill generally seem to work better than more complex ones. There's no difference in the way the models are generated, just the complexity of the shape.

**Steps to Replicate:**
- Using the glitch link below, look vertically down on the model to see where the hotspots are (one in the lake, one in the green area in a valley)
- Change the angle of view so that the lake or green area is no longer visible e.g. blocked from view by one of the big mountains.
- Despite not being able to see the lake / green area, the hotspot will remain visible (solid white filled circle rather than transparent outline only circle)

#### Live Demo

https://glitch.com/~hotspots-not-hidden

### Version

https://unpkg.com/@google/model-viewer@1.3.0/dist/model-viewer.min.js
- model-viewer: v1.3.0

### Browser Affected

- [x] Chrome Version 86.0.4240.193 (Official Build) (64-bit)
- [ ] Edge
- [x] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [x] Windows 10 Home

**Comment by elalish** [comment]
This is actually expected behavior. Doing true 3D occlusion is not very feasible with 2D DOM content, especially not in a performant way. The only way that we do occlusion is by hiding the hotspot when its normal vector points away from the camera, which is simple and fast, but is less than perfect on non-convex models. This is also why we don't hide hotspots completely by default. Our system has the side benefit of being more accessible, since the hotspots retain their full clickable area even when partially occluded.

**Comment by alansegar** [comment]
Thanks for explaining. The behaviour makes sense now and is consistent with how you are describing it's expected to work.
